skopt/plots.py

```diff
@@ -93,322 +98,880 @@ def plot_convergence(*args, **kwargs):
+        if dim.prior == 'log-uniform':
+            a.set_xscale('log')
+
+    # Process the plots below the diagonal.
+    for row in range(n_dims):
+        # Get the search-space dimension for this row.
+        dim_row = dimensions[row]
+
+        # Only iterate until the diagonal.
+        for col in range(row):
+            # Get the search-space dimension for this column.
+            dim_col = dimensions[col]
+
+            # Reference to the plot for this row and column.
+            a = ax[row, col]
+
+            # Plot a grid.
+            a.grid(True)
+
+            # Set the plot-limits to correspond to the search-space bounds.
+            a.set_xlim(dim_col.bounds)
+            a.set_ylim(dim_row.bounds)
+
+            # Use log-scale on the x-axis?
+            if dim_col.prior == 'log-uniform':
+                a.set_xscale('log')
+
+            # Use log-scale on the y-axis?
+            if dim_row.prior == 'log-uniform':
+                a.set_yscale('log')
+
+    # Turn off all plots to the upper-right of the diagonal.
+    for row in range(n_dims):
+        for col in range(row+1, n_dims):
+            ax[row, col].axis("off")
+
+    # Set the designated ylabel for the top-left plot.
+    row = col = 0
+    ax[row, col].set_ylabel(ylabel)
+
+    # Set the dimension-names for the left-most column.
+    col = 0
+    for row in range(1, n_dims):
+        ax[row, col].set_ylabel(dimensions[row].name)
+
+    # Set the dimension-names for the bottom row.
+    row = n_dims - 1
+    for col in range(0, n_dims):
+        ax[row, col].set_xlabel(dimensions[col].name)
+
+    # Remove the y-tick labels for all plots except the left-most column.
+    for row in range(n_dims):
+        for col in range(1, n_dims):
+            ax[row, col].set_yticklabels([])
+
+    # Remove the x-tick labels for all plots except the bottom row.
+    for row in range(n_dims-1):
+        for col in range(n_dims):
+            ax[row, col].set_xticklabels([])
+
+
+def _map_bins(bins, bounds, prior):
+    """
+    For use when plotting histograms.
+    Maps the number of bins to a log-scale between the bounds, if necessary.
 
     Parameters
     ----------
-    * `space` [`Space`]
-        The parameter space over which the minimization was performed.
+    * `bins` [int]
+        Number of bins in the histogram.
+
+    * `bounds` [(int, int)]
+        Tuple or list with lower- and upper-bounds for a search-space dimension.
+
+    * `prior` [str or None]
+        If 'log-uniform' then use log-scaling for the bins,
+        otherwise use the original number of bins.
+
+    Returns
+    -------
+    * `bins_mapped`: [int or np.array(int)]:
+         Number of bins for a histogram if no mapping,
+         or a log-scaled array of bin-points if mapping is needed.
+    """
+
+    if prior == 'log-uniform':
+        # Map the number of bins to a log-space for the dimension bounds.
+        bins_mapped = np.logspace(*np.log10(bounds), bins)
```

Comment:
For python we can't do this so we need to explicitly unpack the bounds :(

Reply:
Is this a Python 2.7 compatibility thing?

Reply:
Fixed. I tested it, it's a Python 2.7 compatibility problem.